src/backend/pyproject.toml

```diff
@@ -137,3 +137,19 @@ python_files = [
     "test_*.py",
     "tests.py",
 ]
+
+[tool.coverage.run]
+branch = true
```

Comment:
I suggest using branch coverage instead of statement coverage, as it seems more comprehensive https://coverage.readthedocs.io/en/latest/branch.html